cpp/src/arrow/compute/kernels/vector_swizzle_test.cc

```diff
@@ -0,0 +1,1065 @@
+        InversePermutation(indices, /*max_index=*/0, /*output_type=*/float32()));
+  }
+  {
+    ARROW_SCOPED_TRACE("Output type string");
+    auto indices = ArrayFromJSON(int32(), "[]");
+    ASSERT_RAISES_WITH_MESSAGE(
+        Invalid,
+        "Invalid: Output type of inverse_permutation must be integer, got string",
+        InversePermutation(indices, /*max_index=*/0, /*output_type=*/utf8()));
+  }
+}
+
+TEST(InversePermutation, DefaultOptions) {
+  {
+    ARROW_SCOPED_TRACE("Default options values");
+    InversePermutationOptions options;
+    ASSERT_EQ(options.max_index, -1);
+    ASSERT_EQ(options.output_type, nullptr);
+  }
+  {
+    ARROW_SCOPED_TRACE("Default options semantics");
+    for (const auto& input_type : kIntegerTypes) {
+      ARROW_SCOPED_TRACE("Input type: " + input_type->ToString());
+      auto indices = ArrayFromJSON(input_type, "[0]");
+      ASSERT_OK_AND_ASSIGN(Datum result, InversePermutation(indices));
+      AssertDatumsEqual(indices, result);
+    }
+  }
+}
+
+template <typename ArrowType>
+class TestInversePermutationSmallOutputType : public ::testing::Test {
+ protected:
+  using CType = typename TypeTraits<ArrowType>::CType;
+
+  std::shared_ptr<DataType> type_singleton() {
+    return TypeTraits<ArrowType>::type_singleton();
+  }
+};
+
+TYPED_TEST_SUITE(TestInversePermutationSmallOutputType, SmallOutputTypes);
+
+TYPED_TEST(TestInversePermutationSmallOutputType, JustEnoughOutputType) {
+  auto output_type = this->type_singleton();
+  int64_t input_length =
+      static_cast<int64_t>(std::numeric_limits<typename TestFixture::CType>::max());
+  auto expected =
+      ArrayFromJSON(output_type, "[" + std::to_string(input_length - 1) + "]");
+  DoTestInversePermutationForInputTypes(
+      kIntegerTypes,
+      [&](const std::shared_ptr<DataType>& input_type) {
+        return ConstantArrayGenerator::Zeroes(input_length, input_type);
+      },
+      /*max_index=*/0, output_type, expected);
+}
+
+TYPED_TEST(TestInversePermutationSmallOutputType, InsufficientOutputType) {
+  auto output_type = this->type_singleton();
+  int64_t input_length =
+      static_cast<int64_t>(std::numeric_limits<typename TestFixture::CType>::max()) + 1;
+  for (const auto& input_type : kIntegerTypes) {
+    ARROW_SCOPED_TRACE("Input type: " + input_type->ToString());
+    auto indices = ConstantArrayGenerator::Zeroes(input_length, input_type);
+    ASSERT_RAISES_WITH_MESSAGE(
+        Invalid,
+        "Invalid: Output type " + output_type->ToString() +
+            " of inverse_permutation is insufficient to store indices of length " +
+            std::to_string(input_length),
+        InversePermutation(indices, /*max_index=*/0, output_type));
+  }
+}
+
+TEST(InversePermutation, Basic) {
+  {
+    ARROW_SCOPED_TRACE("Basic");
+    auto indices = "[9, 7, 5, 3, 1, 0, 2, 4, 6, 8]";
+    std::vector<std::string> indices_chunked{
+        "[]", "[9, 7, 5, 3, 1]", "[0]", "[2, 4, 6]", "[8]", "[]"};
```

Comment:
The repetition of test samples is fine, but you may also want to avoid it e.g. through slicing (or conversely concatenation). As you prefer.

Reply:
Removed repetition, only keeping the chunked case, which will be concatenated on the fly.